docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
+2. **`load()`** method loads the contents of the saved file. It calls **`printFileContents(File)`** which will print out all the stored appointments.
+3. **`load()`** method also calls **`addFileContents`** method.
+4. **`addFileContents`** method parses the saved file and identify stored appointments. Depending on the type of appointments, it will then call either **`readConsultationintoAppointmentList()`** / **`readVaccinationintoAppointmentList()`** / **`readSurgeryintoAppointmentList()`** to add these appointment objects into **`ArrayList<Appointment> APPOINTMENTS`**.
+5. If the file is stored in an incorrect format / has missing details, a **`SniffException`** is thrown.
+
+### Find - Find Implementation
+![img_17.png](img_17.png) <br>
+**Figure 10: Sequence Diagram showing the logical implementation of executeCommand() for the Find Command**
+
+- Find command can be used by the user to find up to 3 categories, **`Appointment ID`**, **`Appointments Type`**, **`Animal Type`**.
+- If user input is not supported by these three find commands, a **`SniffException`** is thrown.
+
+#### findAppointment()
+1. **`findAppointment`** loops through arraylist appointments and checks if matching appointment ID is present.
+2. If matching appointment ID is found, it calls **`toString`** and **`formatPrintList`** to print out appointments to user.
+3. If empty appointment ID is provided by user or appointment ID is in an invalid format, **`SniffException`**, is thrown for both cases.
+4. If no matching appointments are stored, an ui method, **`showUserMessage`** is called.
+
+#### findAnimal()
+1. **`findAnimal`** loops through arraylist appointments and checks if appointments for specified animal type is present.
+2. If matching appointments are found, it calls **`toString`** and **`formatPrintList`** to print out appointments to user.
+3. If no matching appointments are stored, ui method, **`showUserMessage`** is called.
+
+#### findType()
+1. **`findType`** loops through arraylist appointments and checks if specified appointment type (surgery, consultation, vaccination) is present.
+2. If appointments are found, it calls **`toString`** and **`formatPrintList`** to print out appointments to user.
+3. If no matching appointments are stored, ui method, **`showUserMessage`** is called.
+
+### Appointment - Class Implementation (non UID)
+![img_19.png](img_19.png)<br>
+**Figure 9: UML Diagram of Appointment Class**
+1. The appointment class takes in the user input of adding an appointment to the list of appointments.
+2. The **`Appointment class`** named `Appointment` takes in input such as uid, name, type of appointment, animal type, and date of appointment. This appointment will then  be added to the list.
+3. If the user inputs omits any entry or adds any extra entry then an error message will be displayed. An error message will also be displayed if the input type is of the wrong the format.
+
+##### UID Generation
+![img_20.png](img_20.png)<br>
```

Comment:
Perhaps you can consider using the standard notation for class diagrams
![image](https://user-images.githubusercontent.com/88631870/228733887-89a8ae76-1784-415c-8523-e78ba95315c0.png)